### ISSUE TITLE
`Get-Credential` throws exception instead of prompting for input in PowerShell 5.1

### ISSUE THREAD

**Issue by csandfeld** [PowerShell/vscode-powershell#3820]
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all [issues](https://github.com/PowerShell/vscode-powershell/issues?q=is%3Aissue) to ensure it has not already been reported.
- [X] I have read the [troubleshooting](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md) guide.
- [X] I am sure this issue is with the _extension itself_ and does not reproduce in a standalone [PowerShell](https://github.com/PowerShell/PowerShell/issues/new/choose) instance.
- [X] I have verified that I am using the latest version of Visual Studio Code and the PowerShell extension.

### Summary

When using preview v2022.2.0 with Windows PowerShell 5.1, this exception is throw in the integrated console when running Get-Credential.
```PowerShell
PS C:\> Get-Credential
Get-Credential : Cannot process command because of one or more missing mandatory parameters: Credential.
At line:1 char:1
+ Get-Credential
+ ~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Get-Credential], ParameterBindingException
    + FullyQualifiedErrorId : MissingMandatoryParameter,Microsoft.PowerShell.Commands.GetCredentialCommand
 
PS C:\> 
```

This is not the case when using PowerShell Extension v2021.12.0, or when using PowerShell 7.2.1 instead of Windows PowerShell.

### PowerShell Version

```console
PS C:\> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.19041.1320
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.1320
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### Visual Studio Code Version

```console
PS C:\> code --version
1.64.1
d6ee99e4c045a6716e5c653d7da8e9ae6f5a8b03
x64
PS C:\>
```

### Extension Version

```console
PS C:\> code --list-extensions --show-versions                           
ms-vscode.powershell@2021.12.0
ms-vscode.powershell-preview@2022.2.0
PS C:\>
```

### Steps to Reproduce

- Open Visual Studio Code
- Install PowerShell Preview extension
- Configure integrated console to use Windows PowerShell x64
- Run Get-Credential (either directly in console, or by pressing F8 - makes no difference)

**Comment by csandfeld** [comment]
Try just `Get-Credential` (without the `-Message 'ok'` param)

**Comment by csandfeld** [comment]
Same test as above with v2021.12.0

![image](https://user-images.githubusercontent.com/2597673/154434704-bbf998be-294c-4c0f-b3d5-4bdbc9b01877.png)

**Comment by csandfeld** [comment]
Is further repro info needed?

**Comment by JustinGrote** [comment]
@csandfeld so I can repro it as you state, not sure why Get-Credential would have its parameters parsed differently in the PSIC than in regular console. @andschwa @SeeminglyScience any ideas?

**Comment by JustinGrote** [comment]
I noticed that the `Credential` parameter is flagged mandatory in 5.1, but it doesn't follow the usual behavior of showing the mandatory parameter error, instead it hooks into the WPF GUI to prompt for the mandatory value, I'm guessing our Host doesn't support that, and it fails. 7.1 doesn't do this because it always uses a text-based prompt rather than trying a GUI prompt.

**Comment by JustinGrote** [comment]
@andschwa It is because our PS Host is advertising NonInteractive mode. Read-Host has same problem. Is interactive mode possible with the new pipeline engine?

```
PSIC> read-host 'test'
read-host : Windows PowerShell is in NonInteractive mode. Read and Prompt 
functionality is not available.
At line:1 char:1
+ read-host 'test'
+ ~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Read-Host], PSInvalidOperatio 
   nException
    + FullyQualifiedErrorId : InvalidOperation,Microsoft.PowerShell.Commands.ReadH 
   ostCommand
```

**Comment by andschwa** [comment]
Oh, good find! I can dig into this.

**Comment by andschwa** [comment]
So yes, we've been passing `-NonInteractive` to [PowerShell at startup](https://cs.github.com/PowerShell/vscode-powershell/blob/8e5f6b29e0aac2254e6e2a8d2dd57a044856e6fe/src/process.ts#L76). This sets a flag at a really low level within PowerShell itself, which is a bit frightful, _but_ in my basic testing just now it seems like nothing breaks when we remove the flag. Let's give it a try?

**Comment by andschwa** [comment]
Ok so apparently we need `-NonInteractive` to prevent Windows PowerShell from loading its own bundled PSReadLine (this is pending verification, it may no longer be true in PSESv3) and so the previous workaround was that we'd implemented _our own_ `Get-Credential` hence why this is now showing up. Thanks @SeeminglyScience!

**Comment by andschwa** [comment]
Looks like we've got a fix!

**Comment by andschwa** [comment]
The fix for this is out in [PowerShell Preview v2022.2.2](https://marketplace.visualstudio.com/items?itemName=ms-vscode.powershell-preview)!

**Comment by csandfeld** [comment]
That's great news - thanks @andschwa and @JustinGrote
